### PR TITLE
asdf 0.3.0 (new formula)

### PR DESCRIPTION
**Pull request by vicendominguez**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by vicendominguez** [comment]
😢 asdf keeps that structure..... any idea?

File: Formula/asdf.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on "libtool" => :run
+  depends_on "unixodbc" => :run
+
+  def install
+    bash_completion.install "completions/asdf.bash"
+    prefix.install Dir["*"]
+  end
+
+  def caveats; <<-EOS.undent
+    #########################################
```
Comment (dunn) [inline Formula/asdf.rb]: Don't add these lines.  Keep it concise, like [`bash-preexec.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/bash-preexec.rb).

**Comment by dunn** [comment]
I wonder if this formula should be named `asdf-vm`, to avoid confusion with https://common-lisp.net/project/asdf/

File: Formula/asdf.rb (hunk 1 above)
Comment (vicendominguez) [inline Formula/asdf.rb]: @dunn Just to make sure I understand correctly, are you speaking about the `######` lines?

**Comment by vicendominguez** [comment]
Hey @dunn thx for answering......

On the last comment, I don't have any problem to change the name but my main concern, because I don't know the other project, is if we are using the same binary name......any idea?

File: Formula/asdf.rb (hunk 1 above)
Comment (dunn) [inline Formula/asdf.rb]: Yeah, the caveats should just be the two lines of text.

**Comment by dunn** [comment]
> I don't have any problem to change the name but my main concern, because I don't know the other project, is if we are using the same binary name......any idea?

No, that ASDF isn't in Homebrew.  At first I thought this was a PR to add _it_, since it's also a dependency management tool.  But I don't think it will ever be in Homebrew, and it's probably not as well-known as I think.  So on second thought you can leave the name of this formula as-is.

**Comment by vicendominguez** [comment]
👆 Ok Alex, thanks for your help, here we have the correct caveat lines....(I hope)

File: Formula/asdf.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class Asdf < Formula
+  desc "Extendable version manager with support for Ruby, Node.js, Erlang & more"
+  homepage "https://github.com/asdf-vm"
+  url "https://github.com/asdf-vm/asdf/archive/v0.3.0.tar.gz"
+  sha256 "1d986789bfcbef76dc23cc968ca192a03862a9496df65e7cd5d6f8117bc85789"
+
+  bottle :unneeded
+
+  depends_on "coreutils" => :run
```
Comment (dunn) [inline Formula/asdf.rb]: These don't all have to be `:run`; that's only used for dependencies that are normally build-time-only, like automake, autoconf, and libtool.

**Review by dunn — CHANGES_REQUESTED** [review]

**Comment by dunn** [comment]
There's an audit failure too:
```
Error: 1 problem in 1 formula
asdf:
  * Non-executables were installed to "/usr/local/opt/asdf/bin"
    The offending files are:
      /usr/local/opt/asdf/bin/private
```
Can that be installed somewhere else, like `libexec`?

File: Formula/asdf.rb (hunk 2 above)
Comment (vicendominguez) [inline Formula/asdf.rb]: Hello again @dunn,

As you saw, asdf is a version manager which uses some plugins to control the version of the software you want, that version of the software and those plugins requires a build process. I am only following their recommendation, in the [README](https://github.com/asdf-vm/asdf/blob/master/README.md) you can read:

```
For most plugins, it is good if you have installed the following packages OR their equivalent on your OS

OS X: Install these via homebrew coreutils automake autoconf openssl libyaml readline libxslt libtool unixodbc
```

coreutils, automake,  libtool.... are coming from there 😅 . Removing them you will break the main feature imo.... I can not tell something else... what is your recommendation?

Comment (dunn) [inline Formula/asdf.rb]: I mean you can change it to this:
```ruby
  depends_on "autoconf" => :run
  depends_on "automake" => :run
  depends_on "libtool" => :run

  depends_on "coreutils"
  depends_on "libyaml"
  depends_on "openssl"
  depends_on "readline"
  depends_on "unixodbc"

Comment (vicendominguez) [inline Formula/asdf.rb]: got it 👍

**Comment by vicendominguez** [comment]
hey @dunn , all changes done :) In this moment tests are failing but I don't know why

**Comment by dunn** [comment]
```
Error: 2 problems in 1 formula
asdf:
  * Dir["bin/private"] is unnecessary; just use "bin/private"
  * "#{prefix}/lib" should be "#{lib}"

**Comment by vicendominguez** [comment]
🎉  GREEN! 👆

**Review by dunn — APPROVED** [review]

**Comment by dunn** [comment]
Thanks for contributing to Homebrew!  Merged this as 37954ce7325f848f5867b9e320a8e91895d7d619.

**Pull request closed without merging**